src/main/webapp/components/api/cache-creation-controller.service.js

```diff
@@ -0,0 +1,1146 @@
+             * @param segments	INT	Controls the number of hash space segments which is the granularity for key distribution
+             * in the cluster. Value must be strictly positive.
+             * @param indexing STRING	If enabled, entries will be indexed when they are added to the cache.
+             * Indexes will be updated as entries change or are removed.
+             * @param auto-config	BOOLEAN	If enabled, will configure indexing automatically based on cache type
+             * @param statistics	BOOLEAN	If enabled, statistics will be collected for this cache
+             * @param remote-timeout	LONG	In SYNC mode, the timeout (in ms) used to wait for an acknowledgment when making
+             * a remote call, after which the call is aborted and an exception is thrown.
+             * @param capacity-factor	DOUBLE	Controls the proportion of entries that will reside on the local node, compared
+             * to the other nodes in the cluster. Value must be positive. This element is only used in 'distributed' cache instances.
+             * @param batching	BOOLEAN	If enabled, the invocation batching API will be made available for this cache.
+             * @param start	STRING	The cache start mode, which can be EAGER (immediate start) or LAZY (on-demand start).
+             * @param l1-lifespan	LONG	Maximum lifespan of an entry placed in the L1 cache. This element configures the L1 cache
+             * behavior in 'distributed' caches instances. In any other cache modes, this element is ignored.
+             * @param remote-site	STRING	The name of the remote site containing the cache that backups data here.
+             * @param jndi-name	STRING	The jndi-name to which to bind this cache instance.
+             * @param queue-size	INT	In ASYNC mode, this attribute can be used to trigger flushing of the queue when it reaches a specific threshold.
+             * @param async-marshalling	BOOLEAN	If enabled, this will cause marshalling of entries to be performed asynchronously.
+             * @param queue-flush-interval	LONG	In ASYNC mode, this attribute controls how often the asynchronous thread used to flush the
+             * replication queue runs. This should be a positive integer which represents thread wakeup time in milliseconds.
+             * @param owners	INT	Number of cluster-wide replicas for each cache entry.
+             *
+             *
+             */
+            CacheCreationControllerClient.prototype.createDistributedCache = function (address, mode, module_param, remote_cache, segments, indexing,
+                                                                    auto_config, statistics, remote_timeout, capacity_factor,
+                                                                    batching, start, l1_lifespan, remote_site, jndi_name,
+                                                                    queue_size, queue_flush_interval, owners) {
+
+
+              var op = {
```

Comment:
You can either just pass the given config object, or better and more secure: create new one by copying values from it. To copy values you can use for loop.